### PR TITLE
Zephyr multicore fixes

### PR DESCRIPTION
**Pull request by lyakh**
Fix a deadlock with multicore scheduling under Zephyr, fix unprotected domain data, improve per-core locking.

### CODE REVIEW

File: src/schedule/zephyr_ll.c [hunk 1]
```diff
@@ -36,6 +36,16 @@ struct zephyr_ll_pdata {
 	struct k_sem sem;
 };
 
+static void zephyr_ll_lock(struct zephyr_ll *sch, uint32_t *flags)
```
Comment (lgirdwood) [inline src/schedule/zephyr_ll.c:39]: Can we call the Zephyr API directly ?

File: src/schedule/zephyr_domain.c [hunk 2]
```diff
@@ -159,6 +160,8 @@ static int zephyr_domain_register(struct ll_schedule_domain *domain,
 
 	k_thread_start(thread);
 
+	spin_lock_irq(&domain->lock, flags);
```
Comment (lgirdwood) [inline src/schedule/zephyr_domain.c:163]: can we call the zephyr spinlock here too.

File: src/schedule/zephyr_ll.c (hunk 1 above)
Comment (lyakh) [reply src/schedule/zephyr_ll.c:39]: @lgirdwood don't think so, not yet. Zephyr spinlocks use `struct k_spinlock` and XTOS uses `spinlock_t`. But this spinlock is removed in a follow up commit anyway, it's just abstracted here for convenience, for easier replacement and removal.

File: src/schedule/zephyr_domain.c (hunk 2 above)
Comment (lyakh) [reply src/schedule/zephyr_domain.c:163]: @lgirdwood same here - the argument is of a different type and that spinlock is a member of `struct ll_schedule_domain` which is also used with XTOS. We probably want a tree-wide spin-lock migration to Zephyr.

**Comment by lyakh** [comment]
This PR only touches 2 zephyr-related files, not used in QB or in CML_RVP_SDW on-device testing.

File: src/schedule/zephyr_domain.c (hunk 2 above)
Comment (lgirdwood) [reply src/schedule/zephyr_domain.c:163]: we do, pls add that to your tasks for v2.1

**Review by lgirdwood — APPROVED** [review]
@lyakh do we need to update the Zephyr commit ID with the print qualifier update ?

**Comment by lyakh** [comment]
> @lyakh do we need to update the Zephyr commit ID with the print qualifier update ?

@lgirdwood I'm not really sure, but it seems to me we're currently testing with the tip of "main." Let me repush to verify

**Comment by lgirdwood** [comment]
ADL failures on CI not related to Zephyr and looks like a network issues as all tests appear to start correctly but then logger and dmesg stop...

**Pull request merged**